industry_benchmarks/utils/fix_networks.py

```diff
@@ -0,0 +1,630 @@
+    concatenated_edges = old_edges + tape_edges
+    concatenated_network = LigandNetwork(
+        nodes=input_ligand_network.nodes,
+        edges=concatenated_edges,
+    )
+    if not concatenated_network.is_connected():
+        raise ValueError(
+            "The attempt to fix the broken network failed!"
+            "Please contact the OpenFE team if you experience this."
+        )
+
+    return LigandNetwork(nodes= tape_nodes, edges=tape_edges)
+
+
+def get_alchemical_charge_difference(mapping) -> int:
+    """
+    Checks and returns the difference in formal charge between state A and B.
+
+    Parameters
+    ----------
+    mapping : dict[str, ComponentMapping]
+      Dictionary of mappings between transforming components.
+
+    Returns
+    -------
+    int
+      The formal charge difference between states A and B.
+      This is defined as sum(charge state A) - sum(charge state B)
+    """
+    chg_A = Chem.rdmolops.GetFormalCharge(
+        mapping.componentA.to_rdkit()
+    )
+    chg_B = Chem.rdmolops.GetFormalCharge(
+        mapping.componentB.to_rdkit()
+    )
+
+    return chg_A - chg_B
+
+
+def get_settings():
+    """
+    Utility method for getting RFEProtocol settings for non charge changing
+    transformations.
+    These settings mostly follow defaults but use the newest OpenFF 2.2.
+    """
+    # Are there additional settings we should specify here?
+    settings = RelativeHybridTopologyProtocol.default_settings()
+    settings.engine_settings.compute_platform = "CUDA"
+    # Should we use this new OpenFF version or the default?
+    settings.forcefield_settings.small_molecule_forcefield = 'openff-2.2.0'
+    # Only run one repeat per input json file
+    settings.protocol_repeats = 1
+    return settings
+
+
+def get_settings_charge_changes():
+    """
+    Utility method for getting RFEProtocol settings for charge changing
+    transformations.
+
+    These settings mostly follow defaults but use longer
+    simulation times, more lambda windows and an alchemical ion.
+    """
+    settings = RelativeHybridTopologyProtocol.default_settings()
+    settings.engine_settings.compute_platform = "CUDA"
+    # Should we use this new OpenFF version or the default?
+    settings.forcefield_settings.small_molecule_forcefield = 'openff-2.2.0'
+    settings.alchemical_settings.explicit_charge_correction = True
+    settings.simulation_settings.production_length = 20 * unit.nanosecond
+    settings.simulation_settings.n_replicas = 22
+    settings.lambda_settings.lambda_windows = 22
+    # Only run one repeat per input json file
+    settings.protocol_repeats = 1
+    return settings
+
+
+def get_fixed_alchemical_network(ducktape_network, alchemical_network):
+    """
+    Create an alchemical networks with only the missing edges.
+    """
+    solv = openfe.SolventComponent()
+
+    for node in alchemical_network.nodes:
+        prot_comps = [isinstance(comp, ProteinComponent) for comp in node.components.values()]
```

Comment:
```suggestion
        prot_comps = [comp for comp in node.components.values() if isinstance(comp, ProteinComponent)]
```
The code above produces a list of `[False, False, True, False]` which leads to saving of `False` as `ProteinComponent`

Reply:
good catch!